stream/client.go

```diff
@@ -0,0 +1,83 @@
+package stream
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"fmt"
+	"net/http"
+
+	"github.com/google/zoekt"
+	"github.com/google/zoekt/query"
+	"github.com/google/zoekt/rpc"
+)
+
+type client struct {
+	address string
+	path    string
+	conn    *http.Client
+}
+
+func NewClientAtAddress(address string) *client {
+	rpc.RegisterGob()
+	return &client{address, DefaultSSEPath, &http.Client{
+		Transport: &http.Transport{
+			MaxIdleConns: 500,
+		},
+	}}
+}
+
+type Streamer interface {
+	Send(*zoekt.SearchResult)
+}
+
+// Use StreamerChan to cast a receiving channel of search results to a Streamer.
+type StreamerChan chan<- *zoekt.SearchResult
```

Comment:
is this type useful to export? Seems quite easy to implement by the caller of the client. Do we have multiple users of it? If so then probably useful to export and supply for convenience.

Reply:
We use it for the client and for the test and I anticipate we will use it in Sourcegraph when we create the first zoektStreamer. We can also export later if we need it.